### DISCUSSION TITLE
Structured Conversations

### DISCUSSION THREAD

**Discussion by nickoferrall** [ParabolInc/parabol#6223] [Ideas]
During S.P.A. 18, two teams worked on ideas for Structured Conversations. Here are the Notion docs with their final output:
- [Team West](https://www.notion.so/parabol/West-Structured-Conversations-e0940fe3694d40c08768371967139445)
- [Team East](https://www.notion.so/parabol/East-Structured-Conversations-1e36f6057a8d4162a205506080fa784c)

To keep up the momentum and to prevent ideas from being lost in Slack, I wanted to create this discussion so we can jot down notes related to this product. Hopefully, this discussion can help the designers when designing the first version.

**Comment by nickoferrall** [comment]
As discussed during S.P.A., many of us would like to build a “Create your own” meeting type.

Speaking for team East, in this meeting type, a user would use Parabol meeting phases like lego blocks.

For example, to create a 5 Whys meeting, they could add a Reflect phase with one prompt such as “Why is the app down?“. This would be followed by a Group & Vote phase, and would repeat for 5 rounds. They may or may not wish to include a Discuss phase at any point.

In v1, the single prompt in the Reflect phase could be edited by the user. In v2 (or later), we could offer a setting that predicts the prompt title based on the group that got the most votes in the previous round. For example, if the group “GQL executor was unresponsive” got the most votes, the prompt title would be “Why was the GQL executor unresponsive?“.

To create this, we would need to design a “meeting builder”, where the user can add different phases to their meeting. Ideas and initial sketches for this can be found in the Notion docs linked in the description.

**Comment by enriquesanchez** [comment]
Coincidentally, team west also explored the idea of "build your own meeting" and Lego blocks. We looked at the "blocks" Parabol currently supports, and what other types of activities/meetings we could build with them.

We also looked at some popular types of activities, such as governance, 1:1, 5 whys, etc, and the new types of blocks we would need to implement in order to support them.

As for the meeting building aspect of our exploration, we looked at two scenarios: creating a meeting template before hand, and adding "blocks" to a currently open meeting, a "build as you go" kind of thing.

**Reply by avivapinchas** [reply]
To me, the questions that stood out from our exploration were: 

**1. How many new 'components' or 'blocks' or 'lego pieces' do we need to create in order to make many more types of meetings?** 

In the meeting types we explored, we found that 3 things would give us a lot of options: 
- A 'document' meeting phase, where team members can reply to a document they are all viewing
- Additional voting options, especially a veto or objection feature or a ranked-choice voting option 
- Options to repeat cycles 

Of the 9 experiences we considered, 7 could be accomplished with just these changes, alongside what we already offer. 

We also noted that some options may be settings on existing phases or 'blocks'. For example: a 'voting' phase may have the option to be majority wins, ranked choice, objections, etc. These may not need to be different blocks, but options on one blocks. 

**2. Who would create custom meetings, and why?** 

We identified a few scenarios: 
- A facilitator realizes on-the-spot during a meeting that they want another type of interaction, and adds it on the spot 
- A facilitator wants to create a unique experience, ahead of the meeting 

Interestingly: both teams looked at our `add an agenda item` pattern as a way to add new phases within an activity. 

**3. What would be important to offer first, and what restrictions would feel appropriate?**  

We suggested a few things: 
- Entirely new concepts cannot be created: We have meetings, reflections, comments, tasks, etc. Users could not create entirely new concepts, like 'user feedback' or 'proposal' in the context of a custom activity. 
- New activities start from existing ones: We believe that facilitators creating entirely custom experiences would be relatively rare. This feature might make most sense for some kind of 'experts' network, or for paid plans (Enterprise accounts may want to standardize a practice where most Personal accounts probably don't care)
- Sharing, pinning is an opportunity to differentiate between plans: Personal accounts (limited to 2 teams) may not care about sharing a custom meeting across their organization, but this would become valuable for an Enterprise account. Similarly, as more types are created, there may be a need to raise signal on key types. With larger accounts like Enterprise, this feature becomes valuable.

**Reply by nickoferrall** [reply]
> In the meeting types we explored, we found that 3 things would give us a lot of options:
> - A 'document' meeting phase, where team members can reply to a document they are all viewing
> - Additional voting options, especially a veto or objection feature or a ranked-choice voting option
> - Options to repeat cycles

Completely agree, it's exciting that so much can be built using components we already have!
Team East also felt a document phase would be really helpful. 
@tianrunhe had a great idea to reuse our Poker components for the voting options.
I think being able to use a meeting builder to add the option to repeat cycles is the priority for v1. Once that's built, we could ship it and then begin working on the document phase for v2. 

> We believe that facilitators creating entirely custom experiences would be relatively rare

I also think that a facilitator would have to use Parabol for a while before creating custom meetings, but I wonder whether some custom meeting types would be popular right away, for example, 5 Whys? By adding the repeat cycles logic, perhaps we could experiment with adding 5 Whys to the meeting type carousel rather than making users custom build it?

**Reply by avivapinchas** [reply]
> I wonder whether some custom meeting types would be popular right away, for example, 5 Whys? By adding the repeat cycles logic, perhaps we could experiment with adding 5 Whys to the meeting type carousel rather than making users custom build it?

For sure! And I think this could be an opportunity to engage with 'experts' in the same ways tools like Notion do: work with a few folks who do a lot of coaching or facilitating to define some unqiue types of meetings they use regularly, build those into Parabol, and then present that as 'So and So's X meeting'. 

I like the idea of working with some key folks to build these types of things first, vs going straight to a fully `BYO meeting`, because I think we could get there more quickly and we get the added benefit of having that collaboration. 

On the Growth side, we've [started a list of these kinds of folks](https://www.notion.so/parabol/b7b1dd8c59bd4085a05ffb9765b336cb?v=310774b645364283a94ccb4c0f06c4b2), and will be [talking with them to understand what they would want ](https://www.notion.so/parabol/What-would-make-UGC-enticing-for-coaches-users-7fe2fcfc2c4e48cd901bdadbdb154cde), to help us understand what kind of options and customizations we should consider.
